### DISCUSSION TITLE
Char oxidation

### DISCUSSION THREAD

**Discussion by AmirrezaHBSH** [firemodels/fds#13753] [Q&A]
Hello, based on the User Guide, when setting SURFACE_OXIDATION_MODEL=T, FDS uses Eq. 17.8, in which the unit of 'A', as given in Table 17.1, is kg/m²/s. Therefore, based on the same equation, the unit of the char oxidation rate will be kg/m³/s. Consequently, if A's unit is m/s, we need to multiply it by the density of the char to achieve the aforementioned unit and then use it as an input parameter. Is this correct?

However, when SURFACE_OXIDATION_MODEL=T is not set, what will be the unit of A? Additionally, in this case, which equation will FDS use to calculate the rate of char oxidation?

**Comment by ericvmueller** [comment]
A in the surface oxidation model is an empirical coefficient specific to fitting this model and I’m not sure it makes sense to adapt a value obtained from fitting a different model (that has different units).

For oxidation without SURFACE_OXIDATION_MODEL=T see equations 9.1 to 9.3 in the most recent version of the user guide:

https://github.com/firemodels/test_bundles/releases/tag/FDS_TEST

The A value is assumed to have whatever units are required to obtain a rate of kg/m3/s, so it depends on the reaction order (N_S) and temperature dependence (N_T).

**Reply by AmirrezaHBSH** [reply]
> A in the surface oxidation model is an empirical coefficient specific to fitting this model and I’m not sure it makes sense to adapt a value obtained from fitting a different model (that has different units).
> 
> 
> 
> For oxidation without SURFACE_OXIDATION_MODEL=T see equations 9.1 to 9.3 in the most recent version of the user guide:
> 
> 
> 
> https://github.com/firemodels/test_bundles/releases/tag/FDS_TEST
> 
> 
> 
> The A value is assumed to have whatever units are required to obtain a rate of kg/m3/s, so it depends on the reaction order (N_S) and temperature dependence (N_T). 

Thank you for your help, 

So in both cases the unit of reaction rate must be kg/m3/s. Therefore, for both cases with char oxidation model true and false, the reaction rate would be identical. 

My question is, how we need to determine which one should be used? Since in both case using density and surface to volume ratio we can adjust the unit of A value

**Reply by ericvmueller** [reply]
The units of the reaction rate will be identical, yes. The models make different assumptions however. 

The surface_oxidation_model uses the surface oxygen concentration to determine the reaction rate of the whole particle assuming it can be treated as thermally thin. It also constrains N_S=0 and N_O2=1. The default model uses the prescribed in-depth oxygen profile shown in the user guide, and allows the user to set N_S and N_O2. The default is more appropriate if you have a thick solid ...but both are 'research' models and are still being developed and tested.

**Reply by AmirrezaHBSH** [reply]
Thank you for your response. Actually, I have another problem with the equation 17.9 in the user guide about the balance between transport and kinetics. As we discussed before, RHS of the equation unit is kg/m³/s, however, based on the definition of the hm=h/cp, the LHS unit is kg/m2/s. So how we can write the balance between these two?

**Reply by ericvmueller** [reply]
The RHS is also kg/m2/s. Notice the RHS is _not_ the same as 17.8 as the surface-to-volume ratio (1/m) has been factored out.

**Reply by AmirrezaHBSH** [reply]
Thank you, yes I reviewed it again and surface-to-volume ratio (1/m) has been factored out, but in the version that I was using it was not like that. 

In Equation 17.8, the A-value provided in Table 17.1 has units of kg/m2s. However, the A-value I have for my simulation is in m/s. To ensure the reaction rate units in Equation 17.8 are satisfied, I believe the A-value with units of m/s must be multiplied **by the density of char (in kg/m3)** to convert it to kg/m2s. Once converted, I plan to use the SURFACE_OXIDATION_MODEL set to T to model char oxidation. Could you please confirm if this approach is correct?

Thank you for your guidance.

**Reply by ericvmueller** [reply]
I understand this will get you to the correct units, but as I mentioned at the top of the thread I cant confirm this would give you a reasonable behavior. The A for the surface oxidation model assumes the reaction is not dependent on char density but on available surface area. If you are starting with an A with different units then it will have come from fitting an equation of a different form (I'm guessing from Porterie) and it is not obvious to me that you can just re-scale it and use it here. You could see if this gets you reasonable results in an idealized scenario (for example just solving the reaction rate at different temperatures) ...but at a certain point you may as well just tune A to get the result you want. It is a fitting coefficient not a fundamental physical parameter.

**Reply by AmirrezaHBSH** [reply]
> I understand this will get you to the correct units, but as I mentioned at the top of the thread I cant confirm this would give you a reasonable behavior. The A for the surface oxidation model assumes the reaction is not dependent on char density but on available surface area. If you are starting with an A with different units then it will have come from fitting an equation of a different form (I'm guessing from Porterie) and it is not obvious to me that you can just re-scale it and use it here. You could see if this gets you reasonable results in an idealized scenario (for example just solving the reaction rate at different temperatures) ...but at a certain point you may as well just tune A to get the result you want. It is a fitting coefficient not a fundamental physical parameter.

thank you for your reply. I have a question about meshing and cell centered particles. 

&INIT PART_ID              = 'Needles'
      XB                   = 0.1,0.6,0.0,0.08,0.012,0.032
      N_PARTICLES_PER_CELL = 1
      CELL_CENTERED        = .TRUE.
      MASS_PER_VOLUME      = 37.294
      DRY                  = T /

In this line, the XB parameter defines the bed volume containing the pine needles, and MASS_PER_VOLUME represents the bulk density of the simulation, which is calculated based on the pine needle density and the packing ratio.

Using identical parameters, I observe that increasing the mesh resolution results in higher heat flux values at certain points within the bed. I suspect this is related to how the cell-centered approach and the number of particles per cell (N_PARTICLES_PER_CELL) are handled. As the resolution increases, it appears that the number of pine needles per cell also increases, which may lead to higher heat flux measurements at specific points.

Your suggestions on addressing this issue would be greatly appreciated.

**Reply by drjfloyd** [reply]
N_PARTICLES_PER_CELL = 1

The number of particles per cell is not changing. You have fixed this at 1; however, more cells means more total particles (i.e. double the grid cells in each direction and you now have 8 times the number of cells and at 1 particle per cell 8 times the number of particlves but still 1 per cell). Note this does not mean more mass since you have also fixed the mass density (MASS_PER_VOLUME=37.294).  Since you have fixed both density and the number of particles per cell, more resolution will mean more total particles, each cell will still have 1 particle, but that particle will be weighted less (to obtain the correct mass per volume).  

Assuming that you have combustion, more resolution means you are changing the flow field and the resolution of the radiative field. This is going to change how particles heat up and pyrolyze. The fact that increasing the resolution changes the answer doesn't mean there is an issue. If you were poorly resolved before, then increasing the resolution could be making you more resolved.

**Reply by AmirrezaHBSH** [reply]
> N_PARTICLES_PER_CELL = 1
> 
> The number of particles per cell is not changing. You have fixed this at 1; however, more cells means more total particles (i.e. double the grid cells in each direction and you now have 8 times the number of cells and at 1 particle per cell 8 times the number of particlves but still 1 per cell). Note this does not mean more mass since you have also fixed the mass density (MASS_PER_VOLUME=37.294). Since you have fixed both density and the number of particles per cell, more resolution will mean more total particles, each cell will still have 1 particle, but that particle will be weighted less (to obtain the correct mass per volume).
> 
> Assuming that you have combustion, more resolution means you are changing the flow field and the resolution of the radiative field. This is going to change how particles heat up and pyrolyze. The fact that increasing the resolution changes the answer doesn't mean there is an issue. If you were poorly resolved before, then increasing the resolution could be making you more resolved.

Thank you so much for your detailed explanation. In identical simulations, when I use a coarse mesh with dimensions of 6 mm in the X direction, 10 mm in the Y direction, and 6 mm in the Z direction, my results—specifically the bed surface temperature and the heat flux at certain points—closely match the experimental measurements.

However, when I double the mesh size, as you mentioned, the peak values in the plots increase significantly, from 30 kW/m² to 140 kW/m². I was wondering if you could advise if there is anything additional I should include in the input file to mitigate this significant change.

I sincerely appreciate your help.

**Reply by drjfloyd** [reply]
Without knowing your inputs there is no way to suggest anything. Also the fact that your answer changes with the grid size does not mean that there is anything that needs to be corrected. Radiation scales with temperature to the fourth power. If making your grid cells have 1/8 the volume means you now resovle something closer to flame temperatures, then near field radiation can increase a lot.

Predicting fire growth and spread puts you in the realm of research, and you should do some legwork to understand what is happening. You have access to essentially every computed variable via the various outputs. You should attempt to determine why the heat flux is changing. It may be helpful to create simple test problems where you examing the impact of your inputs and the grid size.

**Reply by AmirrezaHBSH** [reply]
> I understand this will get you to the correct units, but as I mentioned at the top of the thread I cant confirm this would give you a reasonable behavior. The A for the surface oxidation model assumes the reaction is not dependent on char density but on available surface area. If you are starting with an A with different units then it will have come from fitting an equation of a different form (I'm guessing from Porterie) and it is not obvious to me that you can just re-scale it and use it here. You could see if this gets you reasonable results in an idealized scenario (for example just solving the reaction rate at different temperatures) ...but at a certain point you may as well just tune A to get the result you want. It is a fitting coefficient not a fundamental physical parameter.

Thank you for your assistance and consideration. I have plotted the total heat flux at a specific point in my pine needle bed, as shown in the attached file. The highest heat flux value should correspond to the combustion reaction, while the second peak is expected to relate to char oxidation, which is exothermic. However, my plot shows a reversed trend.

To analyze this, I defined a solid particle at a specific location and placed the total heat flux device on that solid. Below are my input file details and the resulting plot:


&OBST XB=0.4,0.41,0.04,0.05,0.03,0.04, SURF_ID='solid_particle' /
&SURF ID='solid_particle', COLOR='GRAY',SURFACE_VOLUME_RATIO = 4550, LENGTH= 0.07, MATL_ID='solid_material' /
&MATL ID='solid_material', DENSITY=500.0, CONDUCTIVITY=0.5, SPECIFIC_HEAT=1.5 /
&DEVC ID='hfTOTAL1', XYZ=0.405,0.045,0.035, QUANTITY='TOTAL HEAT FLUX', IOR=1 /

![Capture](https://github.com/user-attachments/assets/d311bf3f-6055-4775-ba26-237f2e15b1e5)


The fire spreads from left to right across my fuel bed, so I set IOR=1. However, I believe the plot might need to be flipped to accurately represent the heat flux values over time. The current plot appears to show an inverted trend, which might not align with the expected progression of heat flux during the fire spread.

Do you have any suggestions for how I can adjust or verify this to ensure the plot correctly reflects the heat flux over time?

**Reply by drjfloyd** [reply]
Without knowing your inputs how are we possibly supposed to make suggestions?

**Reply by AmirrezaHBSH** [reply]
> Without knowing your inputs how are we possibly supposed to make suggestions?

&HEAD CHID='Test_1_0', TITLE='Flame Spread Over Bed of Pine Needles' /

&MISC GVEC=0,0,-9.81 /
&RADI RADIATION=.TRUE. / 

&TIME T_END=450., DT= 0.01/
&MISC TMPA= 27/

&CLIP MINIMUM_DENSITY=0.1 /

&MESH ID='mesh1', IJK=10,17,50, XB=0.0,0.1,0.0,0.17,0.0,0.30, COLOR='BLUE', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh2', IJK=10,17,50, XB=0.1,0.16,0.0,0.17,0.0,0.30, COLOR='RED', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh3', IJK=10,17,50, XB=0.16,0.22,0.0,0.17,0.0,0.30, COLOR='BLUE VIOLET', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh4', IJK=10,17,50, XB=0.22,0.28,0.0,0.17,0.0,0.30, COLOR='BROWN', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh5', IJK=10,17,50, XB=0.28,0.34,0.0,0.17,0.0,0.30, COLOR='COBALT', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh6', IJK=10,17,50, XB=0.34,0.4,0.0,0.17,0.0,0.30, COLOR='DIM GRAY', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh7', IJK=10,17,50, XB=0.4,0.46,0.0,0.17,0.0,0.30, COLOR='FLESH', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh8', IJK=10,17,50, XB=0.46,0.52,0.0,0.17,0.0,0.30, COLOR='GOLD', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh9', IJK=10,17,50, XB=0.52,0.58,0.0,0.17,0.0,0.30, COLOR='GREEN', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh10', IJK=10,17,50, XB=0.58,0.64,0.0,0.17,0.0,0.30, COLOR='HOT PINK', CHECK_MESH_ALIGNMENT=T /
&MESH ID='mesh11', IJK=10,17,50, XB=0.64,0.7,0.0,0.17,0.0,0.30, COLOR='PURPLE', CHECK_MESH_ALIGNMENT=T /



&SURF ID                ='WALL1',
      TMP_FRONT		= 21.0 /

&VENT XB		= 0.0,0.0,0.0,0.17,0.0,0.3,
      SURF_ID		= 'OPEN' /  ! BC left side

&VENT XB		= 0.70,0.70,0.0,0.17,0.0,0.30, 
      SURF_ID		= 'OPEN'/  ! BC right side (pressure outlet)

&VENT XB		= 0.0,0.70,0.0,0.0,0.0,0.30, 
      SURF_ID		= 'OPEN'/
         
&VENT XB		= 0.0,0.70,0.17,0.17,0.0,0.30, 
      SURF_ID		= 'OPEN'/
     
&VENT XB		= 0.0,0.70,0.0,0.17,0.30,0.30, 
      SURF_ID		= 'OPEN',/

&OBST XB		= 0.0,0.7,0.0,0.17,0.0,0.0, 
      SURF_ID		= 'WALL1',
      COLOR		= 'WHITE'
      TRANSPARENCY	= 0.5/

&OBST XB                = 0.0, 0.7, 0.0, 0.08, 0.0, 0.012,
      COLOR             = 'PINK'
      SURF_ID		= 'ASBES'/

&SURF ID        = 'ASBES'
      MATL_ID   = 'ASBESTOS' 
      THICKNESS = 0.012 
      TMP_FRONT = 21.0 /

&MATL ID	    = 'ASBESTOS'
      FYI           = 'Autoclaved aerate concrete'        
      CONDUCTIVITY  = 0.8             
      SPECIFIC_HEAT = 1.0                   
      DENSITY       = 2500.                  
      EMISSIVITY    = 0.80 /

&OBST XB                = 0.1, 0.6, 0.08, 0.081, 0.0, 0.032,
      COLOR             = 'GRAY'
      SURF_ID		= 'FIRECLAY'/

&SURF ID        = 'FIRECLAY'
      MATL_ID   = 'FC' 
      THICKNESS = 0.032 
      TMP_FRONT	= 21.0 /

&MATL ID	    = 'FC'
      FYI           = 'Autoclaved aerate concrete'        
      CONDUCTIVITY  = 0.5            
      SPECIFIC_HEAT = 1.0                   
      DENSITY       = 2500.                  
      EMISSIVITY    = 0.80 /

      

&SURF ID                   = 'PINE_NEEDLE_BED'
      MOISTURE_FRACTION    = 0.009
      SURFACE_VOLUME_RATIO = 4550
      LENGTH               = 0.07
      GEOMETRY             = 'CYLINDRICAL' 
      COLOR                = 'TEAL'
      MATL_ID(1,1)              = 'LIPID'
      MATL_ID(1,2)              = 'DIGEST'
      MATL_ID(1,3)              = 'HEMI_CELLULOSE'
      MATL_ID(1,4)              = 'GLUCAN'
      MATL_ID(1,5)              = 'PHENOLIC'
      MATL_ID(1,6)              = 'ASH'
      MATL_MASS_FRACTION(1,1)   = 0.2273
      MATL_MASS_FRACTION(1,2)   = 0.0990
      MATL_MASS_FRACTION(1,3)   = 0.1744 
      MATL_MASS_FRACTION(1,4)   = 0.201
      MATL_MASS_FRACTION(1,5)   = 0.272
      MATL_MASS_FRACTION(1,6)   = 0.0262 /

&PART ID             	   = 'Needles' 
      DRAG_COEFFICIENT     = 2.81
      COLOR                = 'GREEN'
      SURF_ID              = 'PINE_NEEDLE_BED' 
      STATIC               = .TRUE./
    
      
&INIT PART_ID              = 'Needles'
      XB                   = 0.1,0.6,0.0,0.08,0.012,0.032
      N_PARTICLES_PER_CELL = 1
      CELL_CENTERED        = .TRUE.
      MASS_PER_VOLUME      = 31.429
      DRY                  = T /

&CATF OTHER_FILES='vegetation_model.txt' /

&SURF ID                        = 'IGNITOR',
      TMP_FRONT                 = 1300.,
      COLOR                     = 'BLUE',
      RAMP_T		        = 'RAMP_IGN'/

   
&RAMP ID = 'RAMP_IGN', T = 0.0, F = 1.0 /
&RAMP ID = 'RAMP_IGN', T = 15, F = 1.0 /
&RAMP ID = 'RAMP_IGN', T = 16, F = 0.0 /

&VENT XB = 0.094,0.098,0.0,0.08,0.012,0.012, COLOR = 'BLUE', SURF_ID = 'IGNITOR'/


&OBST XB=0.4,0.41,0.04,0.05,0.03,0.04, SURF_ID='solid_particle' /
&SURF ID='solid_particle', COLOR='GRAY',SURFACE_VOLUME_RATIO = 4550, LENGTH= 0.07, MATL_ID='solid_material' /
&MATL ID='solid_material', DENSITY=500.0, CONDUCTIVITY=0.5, SPECIFIC_HEAT=1.5 /
&DEVC ID='hfTOTAL1', XYZ=0.405,0.045,0.035, QUANTITY='TOTAL HEAT FLUX', IOR=1 /

**Reply by drjfloyd** [reply]
This model cannot be run without your vegetation_model.txt file assuming that has the definition for LIPID, DIGEST, etc.

You have no SLCF outputs for temperature, heat release, etc.
You have no PART outputs for burning rate, heat flux, etc.
Your have only one single DEVC for heat flux to a surface

If you are modeling the growth of fires you are doing research. You need to take the time to try and understand what is happening in your simulation. That cannot be done without having outputs that let you understand what is happening.
